chanfitness/chanevent.go

```diff
@@ -0,0 +1,179 @@
+package chanfitness
+
+import (
+	"time"
+)
+
+type eventType int
+
+const (
+	channelOpenedEvent eventType = iota
+	channelClosedEvent
+	peerOnlineEvent
+	peerOfflineEvent
+)
+
+// String provides string representations of channel events.
+func (e eventType) String() string {
+	switch e {
+	case channelOpenedEvent:
+		return "channel_opened"
+
+	case channelClosedEvent:
+		return "channel_closed"
+
+	case peerOnlineEvent:
+		return "peer_online"
+
+	case peerOfflineEvent:
+		return "peer_offline"
+	}
+
+	return "unknown"
+}
+
+// channelEvent is a a timestamped event which is observed on a per channel basis.
+type channelEvent struct {
+	timestamp time.Time
+	eventType eventType
+}
+
+// chanEventLog stores all events that have occurred over a channel's lifetime.
+type chanEventLog struct {
+	// id is the uint64 of the short channel ID.
+	id uint64
+
+	// peer is the compressed public key of the peer being monitored.
+	peer [33]byte
+
+	// events is a log of timestamped events observed for the channel.
+	events []*channelEvent
```

Comment:
That makes sense, I went for open/close events because I wanted to distinguish between channels that already exist at startup and newly opened ones but I don't think that actually matters? We have an observed lifetime for a channel and observed uptime, open/already existing is external to that.

Reply:
I don't think it matters, because it doesn't show in the reported value in any way.